pkg/apis/apis/v1alpha1/types_apiexport.go

```diff
@@ -186,9 +186,22 @@ const (
 // PermissionClaim identifies an object by GR and identity hash.
 // Its purpose is to determine the added permissions that a service provider may
 // request and that a consumer may accept and allow the service provider access to.
+//
+// +kubebuilder:validation:XValidation:rule="self.all != has(self.resourceSelector)",message="either \"all\" or \"resourceSelector\" must be set"
 type PermissionClaim struct {
 	GroupResource `json:","`
 
+	// all claims all resources for the given group/resource.
+	// This is mutually exclusive with resourceSelector.
+	// +required
+	// +kubebuilder:default=false
+	All bool `json:"all"`
```

Comment:
why not omitempty? IMO having all all the time is ugly.

Reply:
fixed, set `+optional` and `omitemtpy`

Reply:
note: `omitempty` implies `+optional`. I debugged half of a day in controller-tools to find out.